fe1-web/src/features/social/navigation/SocialMediaNavigation.tsx

```diff
@@ -50,58 +48,62 @@ const iconSelector =
 const SocialMediaNavigation = () => {
   const [currentUserPublicKey, setCurrentUserPublicKey] = useState(new PublicKey(''));
 
-  const lao = useSelector(selectCurrentLao);
+  const lao = SocialHooks.useCurrentLao();
 
   if (lao === undefined) {
     throw new Error('LAO is currently undefined, impossible to access to Social Media');
   }
 
   // Get the pop token of the user using the last tokenized roll call
   const rollCallId = lao.last_tokenized_roll_call_id;
-  const eventSelect = makeRollCallSelector(rollCallId);
-  const rollCall: RollCall = useSelector(eventSelect) as RollCall;
+  if (rollCallId === undefined) {
+    throw new Error(
+      'Last tokenized roll call id is undefined, impossible to access to Social Media',
+    );
+  }
+  const rollCall: SocialFeature.RollCall | undefined = SocialHooks.useRollCallById(rollCallId);
 
-  // This will be run again each time the lao.last_tokenized_roll_call_id changes
-  useEffect(() => {
-    generateToken(lao.id, rollCallId)
-      .then((token) => {
-        if (rollCall.containsToken(token)) {
-          setCurrentUserPublicKey(token.publicKey);
-        }
-      })
-      // If an error happens when generating the token, it should not affect the Social Media
-      .catch(() => {
-        /* noop */
-      });
-  }, [lao.id, lao.last_tokenized_roll_call_id, rollCall, rollCallId]);
+  SocialHooks.useSocialContext()
+    .generateToken(lao.id, rollCallId)
+    .then((token) => {
+      if (rollCall?.containsToken(token)) {
+        setCurrentUserPublicKey(token.publicKey);
```

Comment:
I feel like `setCurrentUserPopToken` isn't more explicit since it would imply that the whole token is in state when we only store the public key (the private key is not useful there). It could be `setCurrentUserPopTokenPublicKey` but it feels quite long, isn't it?

Yeah sure, this can be changed. I decided to go for an empty Public Key but an undefined one may be better.